core/src/main/scala/org/apache/spark/SparkContext.scala

```diff
@@ -380,6 +380,10 @@ class SparkContext(config: SparkConf) extends Logging {
         "Deployment to YARN is not supported directly by SparkContext. Please use spark-submit.")
     }
 
+    if (master.startsWith("k8s") && deployMode == "client") {
```

Comment:
Can you do this in `KubernetesClusterManager` instead?

Reply:
Moved the check into `KubernetesClusterManager`.